### PR TITLE
Interactivity

### PR DESCRIPTION
**Pull request by KUBAHO3**
In this project I have:

- added linter setup and fix all linter errors
- added web pack plugins and connected files
- added list interactivity
- added crud operations

To run this project in your system please consider the following practices:

- [] clone this GitHub repository `git clone https://github.com/KUBAHO3/To-Do-List` and navigate to the interactivity branch `cd To-Do-List`, `git checkout interactivity`
- [] install all node packages by `npm install`
- [] run the following command in your terminal to build the system on your dev `npm run build`
- [] Navigate to the `base_url/dist/`

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -54,5 +73,13 @@ taskContainer.addEventListener('click', (e) => {
    }
  });
 
+ clearAllDone.addEventListener('click',(e)=>{
+  const data= Operations.getAllTasks();
+  const storage=data.filter((todo)=>todo.completed===false);
+  console.log(storage)
```
Comment (Qoosim) [inline src/index.js]: - [x] Kindly remove this unnecessary `console.log()`. It does not make your codebase look professional. It is used for debugging.

File: src/index.js [hunk 2]
```diff
@@ -4,15 +4,34 @@ import './style.css';
...
+            description.style.textDecoration = 'none';
+          } else {
+            data[index].completed = true;
+            description.style.textDecoration = 'line-through';
+          }
+          Operations.updateTask(data);
+        }
+      }
+    }
+  });
```
Comment (Qoosim) [inline src/index.js:34]: - [x] Great job so far. There's little issue here, when you mark some tasks completed and click on the `Clear all completed` button to delete all the tasks marked as completed, the tasks get deleted but the indexes of the remaining tasks are not re-ordered so when a new task is added, there's every possibility that two tasks would have the same index number which will make it difficult to mark them completed and save them in the local storage. This is what you need to work on before your project could be approved. Kindly fix that. 

[app-behave.webm](https://user-images.githubusercontent.com/26597432/229086380-a3092c10-a362-4824-a6f2-8130e4ebdf5b.webm)

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @KUBAHO3, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct GitHub Flow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (KUBAHO3) [reply src/index.js]: done

File: src/index.js (hunk 2 above)
Comment (KUBAHO3) [reply src/index.js:34]: resolved as well

**Review by abdulhamiid — APPROVED** [review]
# :green_circle:  Approved! :partying_face:
Hi @KUBAHO3 👋, 
image
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🌟

- ✔️ Descriptions are very descriptive
- ✔️ Linters checks are passing
- ✔️ Use GitHub Flow
- ✔️ Meaningful commit messages.

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**